jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -364,6 +364,13 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
                  const lit_utf8_byte_t *str_p, /**< input string pointer */
                  const lit_utf8_byte_t **out_str_p) /**< [out] matching substring iterator */
 {
+#ifdef REGEXP_RECURSION_LIMIT
+  if (re_ctx_p->recursion_depth >= REGEXP_RECURSION_LIMIT)
```

Comment:
Comparing to 0 is usually easer than comparing to larger constants since the constant might not be represented as immediate values. Furthermore decreasing a number and then comparing the result to 0 is usually free on many cpus (x86, ARM)

So what about changing this into:
```
if (--re_ctx_p->recursion_depth == 0) {
}
```

Reply:
This counter checks for the upper limit of the recursion depth. Initialized with zero and it is increasing faster than decreasing. At this line we are checking the upper limit which is (way more) greater then zero, default to 1000. Thus its always greater than zero or if it is zero, we did not reach the limit.

Reply:
If you think that it would be better to check with zero, i will change the logic to work that way.

Reply:
It seems one of my comments was not added. Now it appeared above your reply.